### PR TITLE
New dependency for `>dev-python/jsonschema-4`

### PR DESCRIPTION
**Pull request by TheChymera**
Closes: https://bugs.gentoo.org/833280

### CODE REVIEW

File: dev-python/uri_template/metadata.xml
Comment (MatthewGentoo) [inline dev-python/uri_template/metadata.xml:14]: `remote-id type="gitlab"` is for gitlab.com. I don't think there's a way to refer to someone else's GitLab instance here :/

```suggestion
```

File: dev-python/uri_template/metadata.xml
Comment (MatthewGentoo) [inline dev-python/uri_template/metadata.xml:11]: You should check with the Python project before you assign a package to them: https://wiki.gentoo.org/wiki/Project:Python#Adding_packages (but it's probably okay)

File: dev-python/uri_template/metadata.xml
Comment (MatthewGentoo) [inline dev-python/uri_template/metadata.xml:7]: You should mark yourself as a proxy, and also add the proxy-maintainers project (you can copy and paste it out of your other packages).

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (MatthewGentoo) [inline dev-python/uri_template/uri_template-1.1.0.ebuild:13]: I'm not too familiar with GitLab's conventions but I think the reason you've had to do the `MY_HASH` stuff is because this is referring to `uri_template-1.1.0.tar.gz` and not `uri_template-v1.1.0.tar.gz`.

If we instead do

```suggestion
SRC_URI="https://gitlab.linss.com/open-source/${PN}/-/archive/v${PV}/${PN}-v${PV}.tar.gz"
```

We can drop `MY_HASH` and do `S="${WORKDIR}"/${PN}-v${PV}`

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (MatthewGentoo) [inline dev-python/uri_template/uri_template-1.1.0.ebuild:18]: We should be able to drop this, since we don't need extra deps to run the tests.

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (MatthewGentoo) [inline dev-python/uri_template/uri_template-1.1.0.ebuild:21]:
```suggestion
```

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (MatthewGentoo) [inline dev-python/uri_template/uri_template-1.1.0.ebuild:27]: Yay, tests!

**Review by MatthewGentoo — COMMENTED** [review]
Good catch, a little confusing that the two different deps have almost the same name (but not a lot we can do about that...)

File: dev-python/uri_template/metadata.xml
Comment (AndrewAmmerlaan) [reply dev-python/uri_template/metadata.xml:11]: > You should check with the Python project before you assign a package to them: https://wiki.gentoo.org/wiki/Project:Python#Adding_packages (but it's probably okay)

It's okay, the package is a direct dep of jsonschema which is already maintained by the Python project.

File: dev-python/uri_template/metadata.xml
Comment (AndrewAmmerlaan) [inline dev-python/uri_template/metadata.xml:15]: We can probably add `<stabilize-allarches/>` here, since the package is all python and shell scripts.

File: dev-python/jsonschema/jsonschema-4.3.3-r1.ebuild
Comment (arthurzam) [inline dev-python/jsonschema/jsonschema-4.3.3-r1.ebuild:23]: Please use here the new syntax, as of [here](https://projects.gentoo.org/python/guide/depend.html#dependencies-conditional-to-python-version) (variant 3)

```suggestion
	' 3.8)
```

File: dev-python/jsonschema/jsonschema-4.4.0-r2.ebuild
Comment (arthurzam) [inline dev-python/jsonschema/jsonschema-4.4.0-r2.ebuild:23]: Also here :)

File: dev-python/uri_template/metadata.xml
Comment (arthurzam) [reply dev-python/uri_template/metadata.xml:14]: Can confirm it won't work

Comment (arthurzam) [reply dev-python/uri_template/metadata.xml:14]: Can confirm it won't work

Comment (TheChymera) [reply dev-python/uri_template/metadata.xml:14]: @MatthewGentoo 

```
chymera@decohost ~/src/gentoo $ ag "gitlab\." */*/metadata.xml | ag remote
app-backup/deja-dup/metadata.xml:28:		<remote-id type="gitlab">https://gitlab.gnome.org/World/deja-dup</remote-id>
app-crypt/adcli/metadata.xml:13:    <remote-id type="gitlab">https://gitlab.freedesktop.org/realmd/adcli</remote-id>
dev-python/uri_template/metadata.xml:13:		<remote-id type="gitlab">gitlab.linss.com/open-source/uri_template/</remote-id>
net-libs/libslirp/metadata.xml:9:		<remote-id type="gitlab">gitlab.freedesktop.org/slirp/libslirp</remote-id>
sci-libs/symmetrica/metadata.xml:22:    <remote-id type="gitlab">https://gitlab.com/sagemath/symmetrica</remote-id>
```

It seems to be used this way. How should I refer to upstream then? They have a mirror on centralized GitHub.

**Comment by TheChymera** [comment]
@MatthewGentoo resolved all issues other than the upstream thing.

File: dev-python/uri_template/metadata.xml
Comment (AndrewAmmerlaan) [inline dev-python/uri_template/metadata.xml:4]:
```suggestion
	<maintainer type="person" proxied="yes> 
```

See here for the details: https://www.gentoo.org/glep/glep-0067.html#package-maintenance

Comment (TheChymera) [reply dev-python/uri_template/metadata.xml]: @AndrewAmmerlaan ok.

File: dev-python/uri_template/metadata.xml
Comment (TheChymera) [reply dev-python/uri_template/metadata.xml]: ok, just removed the upstream then..

**Review by AndrewAmmerlaan — APPROVED** [review]

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (arthurzam) [inline dev-python/uri_template/uri_template-1.1.0.ebuild:7]: Please use here [PEP517 mode](https://projects.gentoo.org/python/guide/distutils.html#basic-use-pep-517-mode)
```suggestion
DISTUTILS_USE_PEP517=setuptools
PYTHON_COMPAT=( pypy3 python3_{8..10} )
inherit distutils-r1
```

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (arthurzam) [inline dev-python/uri_template/uri_template-1.1.0.ebuild:1]: If it is a new ebuild (as it looks to me), it should be copytight starting from this year, meaning just:
```suggestion
# Copyright 2022 Gentoo Authors
```

**Review by arthurzam — CHANGES_REQUESTED** [review]
Looks very nice, great job. Last small changes before approve 😄

**Comment by TheChymera** [comment]
@arthurzam ok.

File: dev-python/jsonschema/jsonschema-4.3.3-r1.ebuild
Comment (thesamesam) [inline dev-python/jsonschema/jsonschema-4.3.3-r1.ebuild:50]: More professional comment if you can please; upstreams may read this

File: dev-python/jsonschema/jsonschema-4.4.0-r2.ebuild
Comment (thesamesam) [inline dev-python/jsonschema/jsonschema-4.4.0-r2.ebuild:41]: Nit: I'd put this above d_e_t.

File: dev-python/jsonschema/jsonschema-4.3.3-r1.ebuild
Comment (TheChymera) [reply dev-python/jsonschema/jsonschema-4.3.3-r1.ebuild:50]: This is all from 8f8324128330b3832102bf3bdb09434afd672363 by @mgorny .
Not sure if he'd like this changed, but in any case it would be a different issue and different commit.

File: dev-python/jsonschema/jsonschema-4.4.0-r2.ebuild
Comment (TheChymera) [reply dev-python/jsonschema/jsonschema-4.4.0-r2.ebuild:41]: This is all from 8f8324128330b3832102bf3bdb09434afd672363 by @mgorny .
Not sure if he'd like this changed, but in any case it would be a different issue and different commit.

**Comment by thesamesam** [comment]
@arthurzam ping

**Comment by chutz** [comment]
The jsonschema changes look fine to me. I will let it up to proxy-maintainers to merge the new package.

**Review by AndrewAmmerlaan — APPROVED** [review]

File: dev-python/uri_template/metadata.xml
Comment (mgorny) [inline dev-python/uri_template/metadata.xml:17]: Plz add remote-id for pypi.

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (mgorny) [inline dev-python/uri_template/uri_template-1.1.0.ebuild:12]: Sounds like you have a `MY_P` candidate here.

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (mgorny) [inline dev-python/uri_template/uri_template-1.1.0.ebuild:17]: Have you really tested it on all these architectures?

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (mgorny) [inline dev-python/uri_template/uri_template-1.1.0.ebuild:20]: EPYTHON should be quoted.

File: dev-python/jsonschema/jsonschema-4.3.3-r1.ebuild
Comment (mgorny) [inline dev-python/jsonschema/jsonschema-4.3.3-r1.ebuild:36]: One tab here.

File: dev-python/jsonschema/jsonschema-4.3.3-r1.ebuild
Comment (mgorny) [reply dev-python/jsonschema/jsonschema-4.3.3-r1.ebuild:50]: It's fine to add trivial changes as part of this commit.

File: dev-python/jsonschema/jsonschema-4.4.0-r2.ebuild
Comment (mgorny) [inline dev-python/jsonschema/jsonschema-4.4.0-r2.ebuild:1]: Why are you adding two new revisions?

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (mgorny) [inline dev-python/uri_template/uri_template-1.1.0.ebuild:1]: You've added fixups to the wrong commit.

**Review by mgorny — CHANGES_REQUESTED** [review]

File: dev-python/jsonschema/jsonschema-4.4.0-r2.ebuild
Comment (TheChymera) [reply dev-python/jsonschema/jsonschema-4.4.0-r2.ebuild:1]: both versions existed when I first submitted the PR. Solved now.

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (TheChymera) [reply dev-python/uri_template/uri_template-1.1.0.ebuild:18]: I have not, but going by what I can test that would mean downgrading jsonschema to ~amd64 only.

**Comment by TheChymera** [comment]
@mgorny all done, except the USE flags. Given that it's a drop-in replacement, python, and jsonschema was marked for all sorts of arches. I think it makes sense to assume it works until bug report shows otherwise.

**Comment by thesamesam** [comment]
You can't make that assumption, needs rekeywording in a bug once merged.

File: dev-python/uri_template/uri_template-1.1.0.ebuild
Comment (mgorny) [reply dev-python/uri_template/uri_template-1.1.0.ebuild:18]: Yes, that's how we handle new packages.

**Comment by TheChymera** [comment]
@thesamesam @mgorny ok, done.

**Review by mgorny — APPROVED** [review]
Thanks. I'll do some final polishing (notably add version number to the commit message), test and merge it.

**Pull request closed without merging**
